analyzers/src/SonarAnalyzer.CSharp/Rules/TooManyParameters.cs

```diff
@@ -46,6 +48,19 @@ public class TooManyParameters : TooManyParametersBase<SyntaxKind, ParameterList
             NodeToDeclarationName.ContainsKey(node.Kind()) && VerifyCanBeChangedBySymbol(node, semanticModel);
 
         protected override int BaseParameterCount(SyntaxNode node) =>
-            (node as ConstructorDeclarationSyntax)?.Initializer?.ArgumentList?.Arguments.Count ?? 0;
+            node switch
+            {
+                ConstructorDeclarationSyntax ctorDeclaration => ctorDeclaration.Initializer?.ArgumentList?.Arguments.Count ?? 0,
+                ClassDeclarationSyntax classDeclaration => RetrieveBasePrimaryConstructorArguments(classDeclaration),
+                _ => 0,
+            };
+
+        private static int RetrieveBasePrimaryConstructorArguments(ClassDeclarationSyntax node)
+        {
+            var type = node.BaseList?.Types.FirstOrDefault();
+            return PrimaryConstructorBaseTypeSyntaxWrapper.IsInstance(type)
+                ? ((PrimaryConstructorBaseTypeSyntaxWrapper)type).ArgumentList.Arguments.Count
```

Comment:
I was not able to come up with a PrimaryConstructorBaseType with a null ArgumentList, hence I removed the null conditional operator `?`. If you don't have any counter-example I think we can merge and see if we have some AD0001 on peach tomorrow.